src/lib/server/account.js

```diff
@@ -111,3 +111,34 @@ export async function getUserinfo({ cookies, locals }) {
 
 	return rows[0];
 }
+
+export async function checkIfUsernameExists(locals, username) {
+	const { rows } = await locals.pool.query({
+		text: 'SELECT EXISTS (SELECT 1 FROM users WHERE username = $1) AS exists',
+		values: [username]
+	});
+
+	const result = rows[0].exists;
+	return result;
+};
+
+export async function checkIfPasswordIsCorrect(locals, username, password) {
+	const { rows } = await locals.pool.query({
+		text: 'SELECT password AS hash FROM users WHERE username = $1',
+		values: [username]
+	});
+
+	const hash = rows[0].hash;
+	const result = await bcrypt.compare(password, hash);
+	return result;
+};
+
+export async function getUserId(locals, username) {
+	const { rows } = await locals.pool.query({
+		text: 'SELECT user_id AS id FROM users WHERE username = $1',
+		values: [username]
+	});
+
+	const id = rows[0].id;
+	return id;
+};
```

Comment:
Idem que pour `checkIfUsernameExists()`
```suggestion
	return rows[0].id;
}
```